src/Agent.Listener/CapabilitiesProvider.cs

```diff
@@ -0,0 +1,291 @@
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener
+{
+    [ServiceLocator(Default = typeof(CapabilitiesProvider))]
+    public interface ICapabilitiesProvider : IAgentService
+    {
+        Task<Dictionary<string, string>> GetCapabilitiesAsync(string agentName, CancellationToken token);
+    }
+
+    public class Capability
+    {
+        public Capability(string name, string tool = null, string[] paths = null)
```

Comment:
We have one use case for this today, right? In general I tend to prefer not adding code unless you need it (eliminate unnecessary maintenance cost, unnecessary review cost, unnecessary readability impact for any dev needing to read the code for any reason). Since this is going to change anyway when the tools concept is fleshed out, i'm not very concerned about it.